Sources/Ollama/Tools.swift

```diff
@@ -0,0 +1,147 @@
+import Foundation
+
+extension Chat {
+    
+    /// A tool provided to the LLM
+    public struct Tool {
+        /// The definition included in the API calls
+        public let definition: ToolDefinition
```

Comment:
Something feels off about the current distinction between `Tool` and `ToolDefinition`. Can't quite put my finger on it, though... Going to chew on this for a bit.

Reply:
The whole tool definition is weird.

With a tool type, which only has a function type of tool. So you have 2 layers for 1 concept there.

This 3rd layer is to link the function with an actual closure.
Though if auto run is being removed this is basically pointless and can be removed.